frontend/src/components/AbstractModal.tsx

```diff
@@ -0,0 +1,74 @@
+import { Dialog, DialogContent, DialogTitle, DialogActions, Typography, makeStyles, IconButton, Button } from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import * as React from 'react';
+
+
+type ModalType =  {
+    title: string;
+    description: string; 
+    confirmText?: string;
+    cancelText?: string;
+    isOpen: boolean;
+    setisOpen: React.Dispatch<React.SetStateAction<boolean>>;
+    handleConfirm?: any;
+    handleCancel?: any;
+};
+
+const useStyles = makeStyles(theme =>({
+    modal:{
+        position:'absolute',
+        top: theme.spacing(5),
+        padding: theme.spacing(2)
+    },
+    modalContent:{
+        textAlign: 'center'
+    },
+    modalAction:{
+        justifyContent: 'center'
+    },
+    modalTitle: {
+        position: 'absolute',
+        top: theme.spacing(2),
+        right: theme.spacing(0.5)
+    }
+}))
+
+
+export default function AbstractModal(props: ModalType) {
+    const classes = useStyles()
+
+    const handleCancel = () => { 
+        props.setisOpen(false)
```

Comment:
Make this prop part of this component's state so the parent doesn't have to worry about managing it

Reply:
If I do this, the parent would have to figure when to display the component and when not to display it. Thought it might be easier by just setting the state on the component itself.

Reply:
I'd argue that the sole jobs of the modal are to display itself and to assign the callbacks (handleConfirm and handleClose). It does not need `setIsOpen`, presumably the calling component will do that in the two callbacks.

Reply:
@rao53 I feel I may have gave you conflicting advice the other day. I spoke with mike for clarification and agree that the parent should control when the modal is opened/closed. So essentially the `setisOpen` callback is redundant like mike mentions above because the other handlers will likely change some state that causes the modal to disappear.

Reply:
Removed the redundant isOpen prop.